marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/artifacts/core.sjs

```diff
@@ -38,39 +38,7 @@ const registeredArtifactTypes = {
     custom: CustomStep
 };
 
-function getTypesInfo() {
-    const typesInfo = [];
-    for (const artifactType of Object.keys(registeredArtifactTypes)) {
-        if (registeredArtifactTypes.hasOwnProperty(artifactType)) {
-            const artifactLibrary = registeredArtifactTypes[artifactType];
-            const updateRoles = artifactLibrary.getPermissions().filter((perm) => perm.capability === 'update').map((perm) => String(perm.roleId));
-            const readRoles = artifactLibrary.getPermissions().filter((perm) => perm.capability === 'read').map((perm) => String(perm.roleId));
-            const currentRoles = xdmp.getCurrentRoles().toArray().map(String);
-            const manageAdminRolesMustAllMatched = ['manage-admin', 'security'].map((roleName) => String(xdmp.role(roleName)));
-            const hasManageAdminAndSecurity = manageAdminRolesMustAllMatched.every((role) => currentRoles.indexOf(role) !== -1);
-            let currentRoleNames = currentRoles.map(roleId => xdmp.roleName(roleId));
-            let userCanUpdate = false;
-            if (currentRoleNames.includes('admin') || hasManageAdminAndSecurity) {
-                userCanUpdate = true;
-            } else {
-                userCanUpdate = updateRoles.some((roleId) => currentRoles.includes(roleId));
-            }
-            const userCanRead = readRoles.some((roleId) => currentRoles.includes(roleId));
-            typesInfo.push({
-                type: artifactType,
-                fileExtension: getArtifactFileExtension(artifactType),
-                directory: getArtifactDirectory(artifactType),
-                nameProperty: artifactLibrary.getNameProperty(),
-                versionProperty: artifactLibrary.getVersionProperty(),
-                userCanUpdate: userCanUpdate,
-                userCanRead: userCanRead
-            });
-        }
-    }
-    return typesInfo;
-}
-
-const entityServiceDrivenArtifactTypes = ['mapping', 'matching', 'merging', 'mastering', 'custom'];
+const entityServiceDrivenArtifactTypes = ['mapping', 'custom'];
```

Comment:
Since matching/merging/mastering don't exist yet, I removed them from this constant. They can be added in once those step types are created.